src/models.rs

```diff
@@ -433,6 +434,14 @@ impl<'a> Uses<'a> {
             Uses::Repository(repo) => repo.git_ref.is_none(),
         }
     }
+
+    pub(crate) fn unpinned_sha(&self) -> bool {
+        let regex = Regex::new("^[a-fA-F0-9]{40}$").unwrap();
+        match self {
+            Uses::Docker(docker) => ! docker.hash.is_none(),
```

Comment:
Clippy tells me that I could use `docker.has.is_some()` here instead, so I'll do that.